exercises/matrix_determinant/python/tests.py

```diff
@@ -0,0 +1,44 @@
+import unittest
+from main import *
+import random
+import numpy as np
+
+def random_matrix(dim):
+    matrix = []
+    for i in range(dim):
+        row = []
+        for j in range(dim):
+            row.append(random.randint(-20,20))
+        matrix.append(row)
+    return matrix
+
+class Testing(unittest.TestCase):
+    def test_det(self):
+        dim = random.randint(1,5)
+        mat = random_matrix(dim)
+        # Checking against NumPy's determinant function
+        self.assertEqual(det(mat),round(np.linalg.det(mat)))
+
+    def test_first_minor(self):
+        mat = [[0,0,0],[0,0,0],[0,0,0]]
+        self.assertEqual(first_minor([[1,1,1,1],
+                                      [1,0,0,0],
+                                      [1,0,0,0],
+                                      [1,0,0,0]],
+                                     0,0),
+                         mat)
+        self.assertEqual(first_minor([[0,0,1,0],
+                                      [1,1,1,1],
+                                      [0,0,1,0],
+                                      [0,0,1,0]],
+                                     1,2),
+                         mat)
+        self.assertEqual(first_minor([[0,0,0,1],
+                                      [0,0,0,1],
+                                      [0,0,0,1],
+                                      [1,1,1,1]],
+                                     3,3),
+                         mat)
```

Comment:
probably worth at least one test that yields a different result to catch any bugs where `first_minor` always returns `0`